commons/com.b2international.index/src/com/b2international/index/revision/StagingArea.java

```diff
@@ -247,7 +249,10 @@ private Commit doCommit(String commitGroupId, long timestamp, String author, Str
 		final Multimap<ObjectId, ObjectId> removedComponentsByContainer = HashMultimap.create();
 		final Multimap<Class<?>, String> deletedIdsByType = HashMultimap.create();
 
-		stagedObjects.forEach((key, value) -> {
+		stagedObjects.entrySet().stream().filter(entry -> !exclusions.contains(entry.getKey().id())).forEach( entry -> {
```

Comment:
Please extract the filtered view to a private method, `getFilteredStagedObjects()`